lib/Sema/BuilderTransform.cpp

```diff
@@ -2477,58 +2480,72 @@ ConstraintSystem::matchResultBuilder(AnyFunctionRef fn, Type builderType,
     return None;
   }
 
-  if (Context.TypeCheckerOpts.ResultBuilderASTTransform) {
-    ResultBuilderTransform transform(*this, fn.getAsDeclContext(), builderType,
-                                     bodyResultType);
-    auto *body = transform.apply(fn.getBody());
+  if (Context.LangOpts.hasFeature(Feature::ResultBuilderASTTransform)) {
+    auto transformedBody = getBuilderTransformedBody(fn, builder);
+    // If this builder transform has not yet been applied to this function,
+    // let's do it and cache the result.
```

Comment:
Is caching and reusing the AST transform a problem if one builder type supports `buildBlock` and another supports `buildPartialResult`? How is that handled?

Reply:
The caching is done per result builder type, so we'd have multiple transforms of the same closure if context requested it.